File: api-gateway/models/product.py
# api-gateway/models/product.py
from typing import List
from pydantic import BaseModel

class Category(BaseModel):
    id: int
    name: str
    image: str
    slug: str

class ProductResponse(BaseModel):
    id: int
    title: str
    price: float
    description: str
    category: Category
    images: List[str]

class ProductsResponse(BaseModel):
    products: List[ProductResponse]
    total: int
    page: int
    limit: int

File: ecommerce-app/src/types/product.ts
export interface Category {
  id: number;
  name: string;
  image: string;
  slug: string;
}

// En el archivo `types/product.ts`
export interface Product {
  _id: string;  // Agregar _id aquí, ya que es el identificador real
  id: string;  // Si deseas mantener la propiedad `id` para otros usos
  title: string;
  slug: string;
  price: number;
  description: string;
  category: Category;
  images: string[];
  stock: number;
}



export interface ProductsResponse { 
  products: Product[];
  total: number;
  page: number;
  limit: number;
} 